poi/src/main/java/org/apache/poi/hssf/model/DrawingManager2.java

```diff
@@ -96,4 +231,119 @@ public EscherDggRecord getDgg() {
+     *               {@link Workbook#PICTURE_TYPE_WMF}, {@link Workbook#PICTURE_TYPE_PICT},
+     *               {@link Workbook#PICTURE_TYPE_JPEG}, {@link Workbook#PICTURE_TYPE_PNG}, or
+     *               {@link Workbook#PICTURE_TYPE_DIB}.
+     * @return Newly allocated picture without any image data.
+     */
+    @Internal
+    public HSSFPictureData allocatePicture(int format) {
+        EscherBlipRecord blipRecord;
+        short escherTag;
+
+        if (format == PICTURE_TYPE_EMF || format == PICTURE_TYPE_WMF) {
+            blipRecord = new EscherMetafileBlip();
+            escherTag = 0;
+        } else {
+            blipRecord = new EscherBitmapBlip();
+            escherTag = (short) 0xFF;
+        }
+
+        blipRecord.setRecordId((short) (EscherBlipRecord.RECORD_ID_START + format));
+        switch (format) {
+            case PICTURE_TYPE_EMF:
+                blipRecord.setOptions(HSSFPictureData.MSOBI_EMF);
+                break;
+            case PICTURE_TYPE_WMF:
+                blipRecord.setOptions(HSSFPictureData.MSOBI_WMF);
+                break;
+            case PICTURE_TYPE_PICT:
+                blipRecord.setOptions(HSSFPictureData.MSOBI_PICT);
+                break;
+            case PICTURE_TYPE_PNG:
+                blipRecord.setOptions(HSSFPictureData.MSOBI_PNG);
+                break;
+            case PICTURE_TYPE_JPEG:
+                blipRecord.setOptions(HSSFPictureData.MSOBI_JPEG);
+                break;
+            case PICTURE_TYPE_DIB:
+                blipRecord.setOptions(HSSFPictureData.MSOBI_DIB);
+                break;
+            default:
+                throw new IllegalStateException("Unexpected picture format: " + format);
+        }
+
+        EscherBSERecord r = new EscherBSERecord();
+        r.setOptions((short) (0x0002 | (format << 4)));
+        r.setBlipTypeMacOS((byte) format);
+        r.setBlipTypeWin32((byte) format);
+        r.setTag(escherTag);
+        r.setBlipRecord(blipRecord);
+
+        addBSERecord(r);
+        return new HSSFPictureData(r);
+    }
+
+    /**
+     * Returns the pictures in the workbook associated with this {@link DrawingManager2}.
+     * <p>
+     * The pictures are ordered as they are stored in the worksheet.
+     *
+     * @return Pictures in the workbook.
+     */
+    @Internal
+    public List<HSSFPictureData> getAllPictures() {
+        ArrayList<HSSFPictureData> pictures = new ArrayList<>(bStoreContainer.getChildCount());
+        for (EscherRecord record : bStoreContainer) {
+            EscherBSERecord bse = (EscherBSERecord) record;
+
+            // We sometimes encounter sheets that have invalid BSE records
+            EscherBlipRecord blip = bse.getBlipRecord();
+            if (blip != null) {
+                pictures.add(new HSSFPictureData(bse));
+            } else {
+                LOGGER.atDebug().log("Encountered BSE record without a BLIP. BSE records must have a BLIP according to the specification.");
```

Comment:
could we provide a link to the specification here as a comment?

Reply:
It wouldn't be my impulse to do so for a couple of reasons. It's referring to a specification within this project, specifically to a class that's used a few lines above. There's no great way of linking in a way that's not already basically immediately obvious

Just to demonstrate my point:

```suggestion
                /** See {@link EscherBSERecord} */ // <--- Just scroll up a few lines and see it yourself instead. This strikes me as redundant.
                LOGGER.atDebug().log("Encountered BSE record without a BLIP. BSE records must have a BLIP according to the specification.");
```

```suggestion
                // See https://poi.apache.org/apidocs/dev/org/apache/poi/ddf/EscherBSERecord.html <--- Why go to a website when you can already see the real thing in the project you're already working in _immediately_
                LOGGER.atDebug().log("Encountered BSE record without a BLIP. BSE records must have a BLIP according to the specification.");
```

Reply:
okay, just that its not intuitive that POI defined what the format of a BSE record is or what a BLIP is anywhere in this code